### PR TITLE
Ensure all sdkv2 test providers have an id and an update method

### PR DESCRIPTION
**Pull request by VenelinMartinov**
This change tweaks the `EnsureProviderIsValid` method in `pulcheck` to always give resources an `id` property and ensure they pass the TF validation with respect to Update methods.

Previously we'd run into trouble with TF validation when a resource needs an Update and doesn't have one or it doesn't need an Update and has one. The rules are non-trivial so I've opted to just make all resources need an Update and ensure they have one.

### CODE REVIEW

**Comment by VenelinMartinov** [comment]
This change is part of the following stack:

- #2720
    - #2721
        - #2723 ◀
            - #2724

<sub>Change managed by [git-spice](https://abhinav.github.io/git-spice/).</sub>

File: pkg/internal/tests/pulcheck/pulcheck.go [hunk 1]
```diff
@@ -75,7 +89,7 @@ func EnsureProviderValid(t T, tfp *schema.Provider) {
 			}
 		}
 
-		if resourceNeedsUpdate(r) && r.UpdateContext == nil {
+		if r.UpdateContext == nil {
```
Comment (iwahbe) [inline pkg/internal/tests/pulcheck/pulcheck.go:92]: Why do we need to add the `if !resourceNeedsUpdate(r) { ... }` block if we remove the conditional.

**Comment by VenelinMartinov** [comment]
Yeah, that's a bit of a weird one. Turns out Terraform is not consistent with its own rules and some valid schemas can not pass validation.

We call `InternalValidate` on all schemas in `pulcheck.EnsureProviderValid`. That function also fills in some reasonable defaults. Unfortunately the TF`InternalValidate` rejects some valid schemas or if the rules from `InternalValidate` are enforced the schemas fail at runtime. Consider this:

```golang
	"test": {
		Type:     schema.TypeSet,
		Optional: true,
		ForceNew: true,
		Computed: true,
		Elem: &schema.Resource{
			Schema: map[string]*schema.Schema{
				"nested": {
					Type:     schema.TypeString,
					Optional: true,
				},
				"computed": {
					Type:     schema.TypeString,
					Optional: true,
					Computed: true,
				},
			},
		},
	},
```

This schema has all its top-level properties as `ForceNew`, so according to `InternalValidate` it `Update` is redundant. This is not correct, as a user might only update the `nested` property which is not `ForceNew`, and doing that fails during runtime.

This behaviour is very non-intuitive and took me a while to figure out that TF Validation has a bug here. We could say that each person who writes a test is responsible for dealing with this and working around this but that seems harder.

I've instead opted to work around the whole mess by adding an Optional `update_prop` to all schemas, which makes sure that `Update` is necessary. Having extra properties in the schema seems like the lesser evil here. I'll leave a note with the explanation.

**Comment by t0yv0** [comment]
Thanks for the explanation, this makes sense! At the very least this needs to be a comment I think on the "update_prop" - that is it's a no-op prop added to make InternalValidate pass.

**Review by t0yv0 — APPROVED** [review]

**Pull request merged**